src/Pos/Explorer/Txp/Toil/Logic.hs

```diff
@@ -132,3 +150,59 @@ getTxRelatedAddrs (UnsafeTx {..}, _, _) undo =
     toSet = HS.fromList . toList
     -- Safe here, because union of non-empty sets can't be empty.
     unionNE lhs rhs = NE.fromList $ toList $ HS.union (toSet lhs) (toSet rhs)
+
+combineBalanceUpdates :: BalanceUpdate -> [(Address, (Sign, Coin))]
```

Comment:
@gromakovsky because an `Address` may be occurring multiple times in both `plusBalance` and `minusBalance`. The conversion to a proper `HashMap` is basically what `combineBalanceUpdates` does.